File: content/snippets/typescript/exercises/advanced/knapsack_problem.ts
const wt = [3,4,5], val=[30,50,60], W=8
const n=wt.length
const dp: number[][] = Array.from({length:n+1},()=>Array(W+1).fill(0))
for(let i=1;i<=n;i++) for(let w=0; w<=W; w++){ dp[i][w]=dp[i-1][w]; if(w>=wt[i-1]) dp[i][w]=Math.max(dp[i][w], dp[i-1][w-wt[i-1]]+val[i-1]) }
console.log(dp[n][W])
export {}

File: content/snippets/python/exercises/advanced/knapsack_problem.py
wt=[3,4,5]
val=[30,50,60]
W=8
n=len(wt)
dp=[[0]*(W+1) for _ in range(n+1)]
for i in range(1,n+1):
    for w in range(W+1):
        dp[i][w]=dp[i-1][w]
        if w>=wt[i-1]: dp[i][w]=max(dp[i][w], dp[i-1][w-wt[i-1]]+val[i-1])
print(dp[n][W])
